### ISSUE TITLE
Trl vllm-server: invalid choice

### ISSUE THREAD

**Issue by Lauch1ng** [wdrink/SimpleAR#24]
Dear Author，
  I follow your instructions strictly in Train.md to build my GRPO env (trl=0.16.0.dev0  transformers=4.47.0.dev0 ). 
    However, when i run the vllm-server using the following command there is an error:
CUDA_VISIBLE_DEVICES=0 trl vllm-serve --model /share/project/code/SimpleAR/ckpts/SimpleAR-0.5B-SFT
[2025-05-19 17:20:54,153] [INFO] [real_accelerator.py:203:get_accelerator] Setting ds_accelerator to cuda (auto detect)
 [WARNING]  async_io requires the dev libaio .so object and headers but these were not found.
 [WARNING]  async_io: please install the libaio-dev package with apt
 [WARNING]  If libaio is already installed (perhaps from source), try setting the CFLAGS and LDFLAGS environment variables to where it can be found.
 [WARNING]  Please specify the CUTLASS repo directory as environment variable $CUTLASS_PATH
 [WARNING]  sparse_attn requires a torch version >= 1.5 and < 2.0 but detected 2.5
 [WARNING]  using untested triton version (3.1.0), only 1.0.0 is known to be compatible
INFO 05-19 17:20:56 __init__.py:190] Automatically detected platform cuda.
usage: trl
TRL CLI: error: argument command: invalid choice: 'vllm-serve' (choose from 'chat', 'dpo', 'env', 'grpo', 'kto', 'sft')

Could you help me with this problem? Have you ever tried to build the env from the instructions you provided and run GRPO training successfully?
Thanks for your time and help!

**Comment by wdrink** [comment]
Hi, sorry for the late reply and I will look into this today, seems like it is related to the trl version

**Comment by wdrink** [comment]
Hi, could you please check the following package versions:

Name: transformers
Version: 4.47.0.dev0
Summary: State-of-the-art Machine Learning for JAX, PyTorch and TensorFlow
Home-page: https://github.com/huggingface/transformers
Author: The Hugging Face team (past and future) with the help of all our contributors (https://github.com/huggingface/transformers/graphs/contributors)
Author-email: transformers@huggingface.co
License: Apache 2.0 License
Requires: filelock, huggingface-hub, numpy, packaging, pyyaml, regex, requests, safetensors, tokenizers, tqdm
Required-by: compressed-tensors, peft, trl, vllm, xgrammar

Name: trl
Version: 0.16.0.dev0
Summary: Train transformer language models with reinforcement learning.
Home-page: https://github.com/huggingface/trl
Author: Leandro von Werra
Author-email: leandro.vonwerra@gmail.com
License: Apache 2.0
Requires: accelerate, datasets, rich, transformers
Required-by:

**Comment by wdrink** [comment]
Package                           Version       Editable project location
--------------------------------- ------------- -------------------------------------
accelerate                        1.7.0
aiohappyeyeballs                  2.6.1
aiohttp                           3.12.12
aiohttp-cors                      0.8.1
aiosignal                         1.3.2
airportsdata                      20250523
annotated-types                   0.7.0
antlr4-python3-runtime            4.13.2
anyio                             4.9.0
args                              0.1.0
astor                             0.8.1
asttokens                         3.0.0
attrs                             25.3.0
av                                14.4.0
blake3                            1.0.5
cachetools                        5.5.2
certifi                           2025.4.26
charset-normalizer                3.4.2
click                             8.2.1
clint                             0.5.1
cloudpickle                       3.1.1
colorful                          0.5.6
compressed-tensors                0.9.1
contourpy                         1.3.2
cycler                            0.12.1
datasets                          3.6.0
decorator                         5.2.1
decord                            0.6.0
deepspeed                         0.14.4
depyf                             0.18.0
dill                              0.3.8
diskcache                         5.6.3
distlib                           0.3.9
distro                            1.9.0
docstring_parser                  0.16
einops                            0.6.1
einops-exts                       0.0.4
executing                         2.2.0
fastapi                           0.115.12
filelock                          3.18.0
flash_attn                        2.7.4.post1
fonttools                         4.58.2
frozenlist                        1.7.0
fsspec                            2025.3.0
ftfy                              6.3.1
gguf                              0.10.0
gitdb                             4.0.12
GitPython                         3.1.44
google-api-core                   2.25.0
google-auth                       2.40.3
googleapis-common-protos          1.70.0
gradio_client                     0.2.9
grpcio                            1.73.0
h11                               0.14.0
hf_transfer                       0.1.9
hf-xet                            1.1.3
hjson                             3.1.0
httpcore                          0.17.3
httptools                         0.6.4
httpx                             0.24.0
huggingface-hub                   0.32.5
idna                              3.10
ijson                             3.4.0
imageio                           2.37.0
importlib_metadata                8.7.0
interegular                       0.3.3
iopath                            0.1.10
ipython                           9.3.0
ipython_pygments_lexers           1.1.1
jedi                              0.19.2
Jinja2                            3.1.6
jiter                             0.10.0
jsonschema                        4.24.0
jsonschema-specifications         2025.4.1
kiwisolver                        1.4.8
lark                              1.2.2
latex2mathml                      3.78.0
latex2sympy2_extended             1.10.1
lightning-utilities               0.14.3
llava                             1.7.0.dev0
llvmlite                          0.44.0
lm-format-enforcer                0.10.11
loguru                            0.7.3
markdown-it-py                    3.0.0
markdown2                         2.5.3
MarkupSafe                        3.0.2
math-verify                       0.7.0
matplotlib                        3.10.3
matplotlib-inline                 0.1.7
mdurl                             0.1.2
mediapy                           1.2.4
mistral_common                    1.6.0
mpmath                            1.3.0
msgpack                           1.1.0
msgspec                           0.19.0
multidict                         6.4.4
multiprocess                      0.70.16
nest-asyncio                      1.6.0
networkx                          3.5
ninja                             1.11.1.4
numba                             0.61.2
numpy                             1.26.1
nvidia-cublas-cu12                12.4.5.8
nvidia-cuda-cupti-cu12            12.4.127
nvidia-cuda-nvrtc-cu12            12.4.127
nvidia-cuda-runtime-cu12          12.4.127
nvidia-cudnn-cu12                 9.1.0.70
nvidia-cufft-cu12                 11.2.1.3
nvidia-curand-cu12                10.3.5.147
nvidia-cusolver-cu12              11.6.1.9
nvidia-cusparse-cu12              12.3.1.170
nvidia-ml-py                      12.575.51
nvidia-nccl-cu12                  2.21.5
nvidia-nvjitlink-cu12             12.4.127
nvidia-nvtx-cu12                  12.4.127
open_clip_torch                   2.32.0
openai                            1.86.0
opencensus                        0.11.4
opencensus-context                0.1.3
opencv-python                     4.11.0.86
opencv-python-headless            4.11.0.86
outlines                          0.0.46
outlines_core                     0.1.26
packaging                         25.0
pandas                            2.3.0
parso                             0.8.4
partial-json-parser               0.2.1.1.post5
peft                              0.4.0
pexpect                           4.9.0
pillow                            11.2.1
pip                               23.0.1
platformdirs                      4.3.8
portalocker                       3.1.1
prometheus_client                 0.22.1
prometheus-fastapi-instrumentator 7.1.0
prompt_toolkit                    3.0.51
propcache                         0.3.2
proto-plus                        1.26.1
protobuf                          6.31.1
psutil                            7.0.0
ptyprocess                        0.7.0
pure_eval                         0.2.3
py-cpuinfo                        9.0.0
py-spy                            0.4.0
pyairports                        2.1.1
pyarrow                           20.0.0
pyasn1                            0.6.1
pyasn1_modules                    0.4.2
pycountry                         24.6.1
pydantic                          2.11.5
pydantic_core                     2.33.2
Pygments                          2.19.1
pyparsing                         3.2.3
python-dateutil                   2.9.0.post0
python-dotenv                     1.1.0
pytorch-lightning                 2.5.1.post0
pytz                              2025.2
PyYAML                            6.0.2
pyzmq                             26.4.0
ray                               2.46.0
referencing                       0.36.2
regex                             2024.11.6
requests                          2.32.4
rich                              14.0.0
rpds-py                           0.25.1
rsa                               4.9.1
safetensors                       0.5.3
scipy                             1.15.3
sentencepiece                     0.2.0
sentry-sdk                        2.29.1
setproctitle                      1.3.6
setuptools                        66.1.1
shortuuid                         1.0.13
shtab                             1.7.2
six                               1.17.0
smart-open                        7.1.0
smmap                             5.0.2
sniffio                           1.3.1
stack-data                        0.6.3
starlette                         0.46.2
svgwrite                          1.4.3
sympy                             1.13.1
tiktoken                          0.9.0
timm                              1.0.15
tokenizers                        0.20.3
torch                             2.5.1
torchaudio                        2.5.1
torchmetrics                      1.7.2
torchvision                       0.20.1
tqdm                              4.67.1
traitlets                         5.14.3
transformers                      4.47.0.dev0   /opt/tiger/whitelist/ARV/transformers
triton                            3.1.0
trl                               0.16.0.dev0   /opt/tiger/whitelist/ARV/trl
typeguard                         4.4.3
typing_extensions                 4.14.0
typing-inspection                 0.4.1
tyro                              0.9.24
tzdata                            2025.2
urllib3                           1.26.20
uvicorn                           0.34.3
uvloop                            0.21.0
virtualenv                        20.31.2
vllm                              0.7.2
wandb                             0.20.1
watchfiles                        1.0.5
wavedrom                          2.0.3.post3
wcwidth                           0.2.13
websockets                        15.0.1
wheel                             0.45.1
wrapt                             1.17.2
xformers                          0.0.28.post3
xgrammar                          0.1.19
xxhash                            3.5.0
yarl                              1.20.1
zipp                              3.23.0
